src/constraint/encoding.ts

```diff
@@ -141,6 +141,39 @@ export const ENCODING_CONSTRAINTS: EncodingConstraintModel[] = [
       return true;
     }
   },{
+    name: 'scalePropertiesSupportedByScaleType',
+    description: 'Scale properties must be supported by correct scale type',
+    properties: SCALE_PROPERTIES.concat([Property.SCALE, Property.TYPE]),
+    allowEnumSpecForProperties: true,
+    strict: true,
+    satisfy: (encQ: EncodingQuery, schema: Schema, opt: QueryConfig) => {
+      if (encQ.scale) {
+        const scale: ScaleQuery = encQ.scale as ScaleQuery;
+        const hasScaleType: boolean = (scale.type !== undefined);
+
+        /**
```

Comment:
Incorrect type of comment

`/** */` is JSDoc -- for describing class / variable or method.

In this case, you wanna use  `//` or `/* */` (We generally use one or more lines of the former.)